blaze/expr/datetime.py

```diff
@@ -26,9 +27,10 @@ def _schema(self):
     def _name(self):
         return '%s_%s' % (self._child._name, self.attr)
 
-    @property
-    def attr(self):
-        return type(self).__name__.lower()
+    @instance
+    class attr(object):
+        def __get__(self, instance, owner):
+            return owner.__name__.lower()
```

Comment:
Why this descriptor implementation rather than the original `@property`?  In what cases is `owner` different from `type(self)` in the original code?